back-end/test/postUsers.js

```diff
@@ -0,0 +1,43 @@
+const server = require('../app.js')
+const chai = require('chai')
+const chaiHttp = require('chai-http')
+const axios = require('axios')
+const sinon = require('sinon')
+var expect = chai.expect
+chai.use(chaiHttp)
+
+describe('POST /tv_users', () => { //update all fields of user info
+  it('should update all user info fields', async () => {
+    const stub = sinon.stub(axios, 'post').resolves({data: {
+      username: 'newUsername', 
+      email: 'newEmail@gmail.com',
+      password: 'newPassword',}})
+    const res = await chai.request(server).post('/tv_users')
+    expect(res.status).to.equal(200)
+    expect(res.body).to.deep.equal({
+      username: 'newUsername', 
+      email: 'newEmail@gmail.com',
+      password: 'newPassword'})
+      stub.restore()
+    })
```

Comment:
This is very much something I just realized, but I think there needs to be a little bit more to these tests to make sure they're testing the route properly (and this goes for the already merged-in PATCH test too). 

Essentially, `stub` only affects the axios call we make inside the route. For a POST call, this should return the user object that is created. In this case in the test, our Mockaroo API creates a user object that'd look something like this:
```
{
id: <some id mockaroo gives us>,
username: 'newUsername',
email: 'newEmail@gmail.com',
password: 'newPassword',
bio: '',
shows: [],
}
```
`bio` and `shows` are initialized by the Mockaroo API. The object in this form should be what's in the `.resolves({data: object_i_just_showed})`

However, when we make a call to this endpoint from the front-end, we're only passing in the following object:
```
{
username: 'newUsername',
password: 'newPassword',
email: 'newEmail@gmail.com',
}
```
This is the request body (the payload, I think is the more technical term?), which we need to specify is that case when we actually make the post request to `localhost:3000/tv_users`. This can be done by altering `await chai.request(server).post('/tv_users')` to have `.send(object_front-end_passes_in)` after the `.post('/tv_users')`(that object being the one with only 3 fields).

So, after doing all this, we'll have to change the assertions we make (the expect stuff). We'll still expect `res.status` to be 200, but we'll expect `res.body` to be exactly what's in the `.resolve` (the first object I wrote out).

In addition to this, there are a few more things we should check. We want to make sure that the smaller object we pass into the post request as a whole (not the one we're stubbing!) is a parameter of the axios function we"re stubbing! We can do this with `sinon` assertions, specifically `calledWith`. It'll look something like this:
`sinon.assert.calledWith(stub, <the mockaroo URL>, <the second object we definied>`)

Essentially, we're testing that the stub (which is replacing that `axios.post` in the code in `app.js`) has the parameters of the Mockaroo API that the post request is being made to, and the object that is being created.

Reply:
yup, thank you, just pushed the edits

Reply:
yup, just pushed the edits